sympy/core/mul.py

```diff
@@ -1264,26 +1266,38 @@ def _eval_is_odd(self):
         is_integer = self.is_integer
 
         if is_integer:
-            r, acc = True, 1
+            #The entire expression is an integer, but it
+            #can be decomposed into fractions (i.e.
+            #numerator and denominator.
+            #The counters keep track of multiples of 2 in
+            #numerator and denominator to determine oddness.
+            count_denom = 0
+            count_num = 0
             for t in self.args:
-                if not t.is_integer:
-                    return None
-                elif t.is_even:
-                    r = False
-                elif t.is_integer:
-                    if r is False:
-                        pass
-                    elif acc != 1 and (acc + t).is_odd:
-                        r = False
-                    elif t.is_odd is None:
-                        r = None
-                acc = t
-            return r
+                if t == S.Half:
+                    count_denom += 1
+                elif isinstance(t,Symbol):
```

Comment:
Are you referring to the restriction regarding the S.Half?  I'm only keeping track of factors of 2 in the numerator and denominator, since we know the whole expression is an integer.  But I see what you're saying, like if someone input an expression with a 1/6, 1/6 has a factor of 2 in the denominator but with how it handles right now, this factor of 2 would not be recognized by the if statement.  I could change it to also include any denominator that is even, so 0 mod 2.  Would this fix what you were referring to?